packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -321,14 +321,17 @@ class ChangeNotifier implements Listenable {
   @mustCallSuper
   void dispose() {
     assert(ChangeNotifier.debugAssertNotDisposed(this));
+    assert(_notificationCallStackDepth == 0 && _reentrantlyRemovedListeners == 0, 'dispose can not be called during listener callbacks');
     assert(() {
       _debugDisposed = true;
       return true;
     }());
     if (kFlutterMemoryAllocationsEnabled && _creationDispatched) {
       MemoryAllocations.instance.dispatchObjectDisposed(object: this);
     }
+    print('dispose is called $_notificationCallStackDepth $_reentrantlyRemovedListeners, _debugDisposed $_debugDisposed');
     _listeners = _emptyListeners;
+    _reentrantlyRemovedListeners = 0;
```

Comment:
with this, it won't crash in release build

Reply:
Maybe leave that as a comment here for future readers?

Reply:
surely it would still cause a crash? the `notifyListeners` method is going to call `_listeners[i]` with values of `i` bigger than `_emptyListeners.length` if it's in the middle of dispatching listeners.